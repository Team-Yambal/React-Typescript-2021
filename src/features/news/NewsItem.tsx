import React from 'react'
import { NewsArticle } from '../../app/models/News'

export type NewsItemProps = {
  article: NewsArticle
}

export const NewsItem = ({ article }: NewsItemProps) => {
  const date = new Date(article.publishedAt)
  return (
    <article>
      <h1>{article.title}</h1>
      <p>{article.description}</p>
      <img src={article.urlToImage} />
      <a href={article.url}>{article.url}</a>
      <div>
        {article.author}@{article.source.name}
      </div>
      <div>{date.toString()}</div>
    </article>
  )
}
