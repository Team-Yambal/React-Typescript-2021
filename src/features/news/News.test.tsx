import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import { NewsItem } from './NewsItem'

const article = {
  source: {
    id: undefined,
    name: 'Automaton-media.com',
  },
  author: 'Aki Nogishi',
  title:
    '『FF14』第69回PLL新情報まとめ。日本データセンター再編により、光の戦士たちはこの夏“クラス替え” - AUTOMATON',
  description:
    'スクウェア・エニックスは3月4日、『ファイナルファンタジーXIV』の新情報を配信する生放送「第69回プロデューサーレターLIVE」を配信した。その内容をまとめよう。',
  url: 'https://automaton-media.com/articles/newsjp/20220304-194356/',
  urlToImage:
    'https://automaton-media.com/wp-content/uploads/2022/03/20220304-194356-header.jpg',
  publishedAt: '2022-03-04T13:32:13Z',
  content:
    '34XIVFF1469LIVE6.1Part16.16.16.04\r\nNPC\r\n69PLL \r\n6.0\r\n6.156.15UI\r\n6.16.156.1x\r\n4NPC6.1NPC6.12.0\r\nNPC\r\nPLL81\r\n84AI344 1\r\n3\r\n6.0121\r\n6.126.115.33.3\r\nPvPPvP6.1PvPPvPUIPLL\r\nPLL\r\nUI6.06.16.1\r\n206.2\r\n6.18FC… [+122 chars]',
}

test('NewsItem', () => {
  // Snapshot
  const component = renderer.create(<NewsItem article={article} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  // Dom testing
  render(<NewsItem article={article} />)
  const count = screen.getByText(/Aki Nogishi/i)
  expect(count).toBeTruthy()
})
