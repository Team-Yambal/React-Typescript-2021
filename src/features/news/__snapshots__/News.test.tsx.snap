// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItem 1`] = `
<a
  className="sc-bdvvtL sc-iIUQWv eRSpvD"
  href="https://automaton-media.com/articles/newsjp/20220304-194356/"
  target="_news"
>
  <article
    className="sc-eCImPb bwPaRL"
  >
    <div
      className="sc-cxpSdN cFCMKo"
    />
    <div
      className="sc-cxpSdN dhFRpC"
    >
      <h4
        className="sc-jcFjpl"
      >
        『FF14』第69回PLL新情報まとめ。日本データセンター再編により、光の戦士たちはこの夏“クラス替え” - AUTOMATON
      </h4>
      <p
        className="sc-eLwHnm GFANF"
      >
        スクウェア・エニックスは3月4日、『ファイナルファンタジーXIV』の新情報を配信する生放送「第69回プロデューサーレターLIVE」を配信した。その内容をまとめよう。
      </p>
      <div
        className="sc-cxpSdN jzrvpE"
      >
        Aki Nogishi
         : 
        Fri Mar 04 2022 22:32:13 GMT+0900 (日本標準時)
      </div>
    </div>
  </article>
</a>
`;
